internal/gcs/process.go

```diff
@@ -0,0 +1,173 @@
+package gcs
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/Microsoft/hcsshim/internal/cow"
+	"github.com/Microsoft/hcsshim/internal/logfields"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	hrNotFound = 0x80070490
+)
+
+type Process struct {
```

Comment:
Can you add the type assertions for `cow.Process` and `cow.Container` it makes it 100% easier for my brain to know what the intent of these methods are